ci: Windows build fails intermittently when downloading (Failed to download from mirror set)

It seems that sometimes a download will fail, but succeeds on the retry, however the full build will still be marked as failed:

https://cirrus-ci.com/task/4800755426131968?logs=build#L31

```
  A suitable version of powershell-core was not found (required v7.1.0). Downloading portable powershell-core v7.1.0...
  Downloading powershell-core...
    https://github.com/PowerShell/PowerShell/releases/download/v7.1.0/PowerShell-7.1.0-win-x86.zip -> C:\Users\ContainerAdministrator\AppData\Local\Temp\vcpkg\downloads\PowerShell-7.1.0-win-x86.zip
  Failed to download from mirror set:
  https://github.com/PowerShell/PowerShell/releases/download/v7.1.0/PowerShell-7.1.0-win-x86.zip: WinHttpReceiveResponse() failed: 12030
  
C:\Users\ContainerAdministrator\AppData\Local\Temp\vcpkg\scripts\buildsystems\msbuild\vcpkg.targets(94,5): error MSB3073: The command ""C:\Users\ContainerAdministrator\AppData\Local\Temp\vcpkg\scripts\buildsystems\msbuild\..\..\..\vcpkg.exe" install  --x-wait-for-lock --triplet "x64-windows-static" --vcpkg-root "C:\Users\ContainerAdministrator\AppData\Local\Temp\vcpkg\scripts\buildsystems\msbuild\..\..\..\\" "--x-manifest-root=C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\build_msvc\\" "--x-install-root=C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\build_msvc\vcpkg_installed\\" " exited with code 1. [C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\build_msvc\libbitcoinconsensus\libbitcoinconsensus.vcxproj]
  Waiting to take filesystem lock on C:\Users\ContainerAdministrator\AppData\Local\Temp\vcpkg\.vcpkg-root...
  C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\build_msvc
  Installing vcpkg dependencies to C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\build_msvc\vcpkg_installed\
  "C:\Users\ContainerAdministrator\AppData\Local\Temp\vcpkg\scripts\buildsystems\msbuild\..\..\..\vcpkg.exe" install  --x-wait-for-lock --triplet "x64-windows-static" --vcpkg-root "C:\Users\ContainerAdministrator\AppData\Local\Temp\vcpkg\scripts\buildsystems\msbuild\..\..\..\\" "--x-manifest-root=C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\build_msvc\\" "--x-install-root=C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\build_msvc\vcpkg_installed\\" 
  Detecting compiler hash for triplet x64-windows-static...
  A suitable version of powershell-core was not found (required v7.1.0). Downloading portable powershell-core v7.1.0...
  Downloading powershell-core...
    https://github.com/PowerShell/PowerShell/releases/download/v7.1.0/PowerShell-7.1.0-win-x86.zip -> C:\Users\ContainerAdministrator\AppData\Local\Temp\vcpkg\downloads\PowerShell-7.1.0-win-x86.zip
  Extracting powershell-core...

--- a/.cirrus.yml
+++ b/.cirrus.yml
@@ -86,6 +86,7 @@ task:
     PATH: 'C:\jom;C:\Python39;C:\Python39\Scripts;C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\MSBuild\Current\Bin;%PATH%'
     PYTHONUTF8: 1
     CI_VCPKG_TAG: '2021.05.12'
+    VCPKG_DOWNLOADS: 'C:\Users\ContainerAdministrator\AppData\Local\vcpkg\downloads'
     VCPKG_DEFAULT_BINARY_CACHE: 'C:\Users\ContainerAdministrator\AppData\Local\vcpkg\archives'
     QT_DOWNLOAD_URL: 'https://download.qt.io/official_releases/qt/5.12/5.12.11/single/qt-everywhere-src-5.12.11.zip'
     QT_LOCAL_PATH: 'C:\qt-everywhere-src-5.12.11.zip'
@@ -121,6 +122,12 @@ task:
       - ..\configure -release -silent -opensource -confirm-license -opengl desktop -no-shared -static -static-runtime -mp -qt-zlib -qt-pcre -qt-libpng -no-libjpeg -nomake examples -nomake tests -nomake tools -no-dbus -no-libudev -no-icu -no-gtk -no-opengles3 -no-angle -no-sql-sqlite -no-sql-odbc -no-sqlite -no-libudev -no-vulkan -skip qt3d -skip qtactiveqt -skip qtandroidextras -skip qtcanvas3d -skip qtcharts -skip qtconnectivity -skip qtdatavis3d -skip qtdeclarative -skip qtdoc -skip qtgamepad -skip qtgraphicaleffects -skip qtimageformats -skip qtlocation -skip qtmacextras -skip qtmultimedia -skip qtnetworkauth -skip qtpurchasing -skip qtquickcontrols -skip qtquickcontrols2 -skip qtscript -skip qtscxml -skip qtsensors -skip qtserialbus -skip qtserialport -skip qtspeech -skip qtvirtualkeyboard -skip qtwayland -skip qtwebchannel -skip qtwebengine -skip qtwebsockets -skip qtwebview -skip qtx11extras -skip qtxmlpatterns -no-openssl -no-feature-sql -no-feature-sqlmodel -prefix %QTBASEDIR%
       - jom
       - jom install
+  vcpkg_tools_cache:
+    folder: '%VCPKG_DOWNLOADS%\tools'
+    reupload_on_changes: false
+    fingerprint_script:
+      - echo %CI_VCPKG_TAG%
+      - msbuild -version
   vcpkg_binary_cache:
     folder: '%VCPKG_DEFAULT_BINARY_CACHE%'
     reupload_on_changes: true

